### PR TITLE
Add initial support for TEI on HPU

### PR DESCRIPTION
**Pull request by ltowarek**
Main changes compared to the upstream:
* Enabled python backend with support for HPU
* Added Habana PT as a base image
* Updated python requirements to match Habana PT image
  * Removed torch
  * Added transformers

### CODE REVIEW

**Comment by astachowiczhabana** [comment]
LGTM

File: backends/python/server/requirements.txt [hunk 1]
```diff
@@ -34,7 +34,7 @@ requests==2.31.0 ; python_version >= "3.9" and python_version < "3.13"
 safetensors==0.3.3 ; python_version >= "3.9" and python_version < "3.13"
 setuptools==68.2.0 ; python_version >= "3.9" and python_version < "3.13"
 sympy==1.12 ; python_version >= "3.9" and python_version < "3.13"
-torch==2.0.1 ; python_version >= "3.9" and python_version < "3.13"
+transformers==4.35.0 ; python_version >= "3.9" and python_version < "3.13"
```
Comment (libinta) [inline backends/python/server/requirements.txt:37]: this seems old one, do you think it can be upgraded to 4.37.2 ?

File: README.md [hunk 2]
```diff
@@ -1,455 +1,30 @@
...
-
-#### Text Embeddings
-
-You can use any JinaBERT model with Alibi or absolute positions or any BERT, CamemBERT, RoBERTa, or XLM-RoBERTa model
-with absolute positions in `text-embeddings-inference`.
-
-**Support for other model types will be added in the future.**
-
-Examples of supported models:
-
```
Comment (libinta) [inline README.md:69]: can we also have the supported models like below?

Comment (ltowarek) [reply README.md:69]: Sure, what about doing it in the same commit that will upgrade transformers? This way we will retest new transformers package and double check supported models at the same time.

**Comment by libinta** [comment]
Ok.then let's do in a different PR.

**Review by astachowiczhabana — APPROVED** [review]
Change passes basic smoke tests

**Pull request merged**